website_logo/controllers/main.py

```diff
@@ -0,0 +1,68 @@
+
+import functools
+import logging
+import io
+import base64
+
+from odoo import _, http, SUPERUSER_ID
+from odoo.http import request
+from odoo.modules import get_module_resource, registry
+
+db_monodb = http.db_monodb
+_logger = logging.getLogger(__name__)
+
+
+class Website(http.Controller):
+
+    def _image_logo_get(self, cr, domain=None):
+        if domain:
+            cr.execute("""SELECT logo, write_date
+                            FROM website
+                           WHERE domain = %s
+                       """, (domain,))
+        else:
+            cr.execute("""SELECT logo, write_date
+                            FROM website
+                       """)
+        row = cr.fetchone()
+        if row and row[0]:
+            return io.BytesIO(base64.b64decode(row[0])), row[1]
+        return False, False
+
+    @http.route([
+        '/web/binary/website_logo',
+        '/website_logo',
+        '/website_logo.png',
+    ], type='http', auth="none", cors="*")
+    def website_logo(self, dbname=None, **kw):
```

Comment:
question: do we really need this whole machinery just to get the logo?
Why can't we simply get current website from request and grab logo value?

Reply:
We can use current website method and obtain the current website, but this method, if it not find a website that has the current domain, return the firs website un database. But, with the current code, if it not find a website tha has the current domain, it try to find a website that has the 'localhost' domain. That is a differece. If it doesn't matter, we can use the method.